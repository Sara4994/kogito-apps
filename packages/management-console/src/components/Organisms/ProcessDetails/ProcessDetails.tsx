import {TimeAgo} from '@n1ru4l/react-time-ago';
import {Button, Card, CardBody, CardHeader, Form, FormGroup, Text, TextVariants} from '@patternfly/react-core';
import React from 'react';
import {Link} from "react-router-dom";

<<<<<<< HEAD
const ProcessDetails = ({ loading, data }) => {
  const DetailsStyle = {
    marginLeft: '2em',
    height: '25em'
  };
  return (
    <Card style={DetailsStyle}>
      <CardHeader>Details</CardHeader>
      <CardBody>
        <Grid gutter="md">
          <GridItem span={6}>
            <TextContent>
              <Text component={TextVariants.h4}>
                Defintion Id
              </Text>
              <Text component={TextVariants.h4}>Instance State</Text>
            </TextContent>
          </GridItem>
          <GridItem span={6}>
            <TextContent>
              {!loading ? (
                data.ProcessInstances.map(item => {
                  return (
                    <div key={item.id}>
                      <Text component={TextVariants.h4}>{item.processId}</Text>
                      <Text component={TextVariants.h4}>{item.state}</Text>
                    </div>
                  );
                })
              ) : (
                <Text component={TextVariants.h4}>Loading...</Text>
              )}
            </TextContent>
          </GridItem>
        </Grid>
      </CardBody>
      <CardFooter>
        <Button variant="primary" style={{ float: 'right' }}>
          Primary
        </Button>
      </CardFooter>
    </Card>
  );
=======
const ProcessDetails = ({loading, data}) => {
    return (
        <Card>
            <CardHeader>Details</CardHeader>
            <CardBody>
                <Form>
                    <FormGroup label="Name" fieldId="name">
                        <Text component={TextVariants.p}>{data.ProcessInstances[0].processName}</Text>
                    </FormGroup>
                    <FormGroup label="State" fieldId="state">
                        <Text component={TextVariants.p}>{data.ProcessInstances[0].state}</Text>
                    </FormGroup>
                    <FormGroup label="Id" fieldId="id">
                        <Text component={TextVariants.p}>{data.ProcessInstances[0].id}</Text>
                    </FormGroup>
                    <FormGroup label="Endpoint" fieldId="endpoint">
                        <Text component={TextVariants.p}>{data.ProcessInstances[0].endpoint}</Text>
                    </FormGroup>
                    <FormGroup label="Start" fieldId="start">
                        <Text component={TextVariants.p}>
                            {data.ProcessInstances[0].start ?
                                <TimeAgo date={new Date(`${data.ProcessInstances[0].start}`)} render={({error, value}) =>
                                    <span>{value}</span>}/>
                                : ''}
                        </Text>
                    </FormGroup>
                    <FormGroup label="End" fieldId="end">
                        <Text component={TextVariants.p}>
                            {data.ProcessInstances[0].end ?
                                <TimeAgo date={new Date(`${data.ProcessInstances[0].end}`)} render={({error, value}) =>
                                    <span>{value}</span>}/>
                                : ''}
                        </Text>
                    </FormGroup>
                    {data.ProcessInstances[0].parentProcessInstanceId ?
                        <FormGroup label="Parent Process" fieldId="parent">
                            <Text component={TextVariants.p}>
                                <Link to={'/ProcessInstances/' + data.ProcessInstances[0].parentProcessInstanceId}>
                                    <Button variant="secondary">{data.ProcessInstances[0].parentProcessInstanceId}</Button>
                                </Link>
                            </Text>
                        </FormGroup> : ''}
                    {data.ProcessInstances[0].childProcessInstanceId.length>0 ?
                        <FormGroup label="Sub Processes" fieldId="parent">
                            {data.ProcessInstances[0].childProcessInstanceId.map((child, index) =>
                                <Text component={TextVariants.p} key={child} style={{marginTop: '5px'}}>
                                    <Link to={'/ProcessInstances/' + child}>
                                        <Button variant="secondary">{child}</Button>
                                    </Link>
                                </Text>)
                            }
                        </FormGroup> : ''}
                </Form>
            </CardBody>
        </Card>
    );
>>>>>>> 56b1a579d538bf04b73145e896e9e73b9d06d2c3
};

export default ProcessDetails;
