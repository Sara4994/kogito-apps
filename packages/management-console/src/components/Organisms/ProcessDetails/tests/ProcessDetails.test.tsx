import React from 'react';
import { shallow } from 'enzyme';
import ProcessDetails from '../ProcessDetails';

const props = {
    loading: false,
    data: {
        ProcessInstances: [{
            id: '',
            processId: '',
            state: ''
        }]
    }
}

const props2 = {
    loading: true,
    data: {
        ProcessInstances: [{
            id: '2d962eef-45b8-48a9-ad4e-9cde0ad6af88',
            processId: 'hotelBooking',
            state: 'COMPLETED'
        }]
    }
}

describe('Process Details component', () => {
    it('Snapshot tests', () => {
        const wrapper = shallow(<ProcessDetails {...props}/>);
        expect(wrapper).toMatchSnapshot();
      });
    it('Sample test case', () => {
        const wrapper = shallow(<ProcessDetails {...props2}/>);
        expect(wrapper.find('Text').first().prop('component')).toBe('h4');
    });
})

