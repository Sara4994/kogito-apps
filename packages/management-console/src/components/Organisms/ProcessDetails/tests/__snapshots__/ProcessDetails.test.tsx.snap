// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Process Details component Snapshot tests 1`] = `
<Card
  style={
    Object {
      "height": "25em",
      "marginLeft": "2em",
    }
  }
>
  <CardHeader>
    Details
  </CardHeader>
  <CardBody>
    <Grid
      gutter="md"
    >
      <GridItem
        span={6}
      >
        <TextContent>
          <Text
            component="h4"
          >
            Defintion Id
          </Text>
          <Text
            component="h4"
          >
            Instance State
          </Text>
        </TextContent>
      </GridItem>
      <GridItem
        span={6}
      >
        <TextContent>
          <div
            key=""
          >
            <Text
              component="h4"
            />
            <Text
              component="h4"
            />
          </div>
        </TextContent>
      </GridItem>
    </Grid>
  </CardBody>
  <CardFooter>
    <Component
      style={
        Object {
          "float": "right",
        }
      }
      variant="primary"
    >
      Primary
    </Component>
  </CardFooter>
</Card>
`;
